### DISCUSSION TITLE
Is there a way to represent 'x' and 'y' on together instead ?

### DISCUSSION THREAD

**Discussion by hamadah10** [OpenModelica/OpenModelica#11042] [General]
Hello,

I have been using modelica for simple and little projects for the past few weeks. Now, I do have a bigger project that has been stressing me out for a while as I ran into so many issues in the past. Now, I figured out most of the issues. 

Yet, there is something I need to refine the model I created. 

I am using a constant which feeds its input to a signal controller. (PID) What I am trying to model uses two 'constants' to represent a location or a point on an X,Y plane. Now, How can  I do that in modelica? If possible. 

I did some research and found that you can use the block 'add' but that doesn't make sense. I am trying to represent something like (1,1) which doesn't = 1+1 like it was suggested. the model I have created currently works and simulates fine but I'm only using one constant which doesn't capture the other aspect I need? any suggestions that could help out? thank you everyone

**Comment by AnHeuermann** [comment]
If you want to group some variables you are probably looking for records: https://mbe.modelica.university/behavior/equations/record_def/

```modelica
record MyCoordinate
  Real x "X coordinate";
  Real y "Y coordinate";
end MyCoordinate;

constant MyCoordinate p = MyCoordinate(x=1.0, y=-1.0);
```

**Comment by casella** [comment]
You can also use arrays. Instead of instantiating a (scalar) constant generator, you can instantiate a N-dimensional array of them. The output y will become an array output.
